spec/views/recipe_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+RSpec.describe Recipe, type: :view do
+  before :each do
+    @user = User.create(name: 'Chia', email: 'chia@dev.co', password: 'password', password_confirmation: 'password')
+
+    @user1 = User.create(name: 'Eric', email: 'eric@dev.co', password: 'password', password_confirmation: 'password')
+
+    @food1 = Food.create(name: 'ndole', measurement_unit: 'Litter', price: '1 hour', user_id: @user.id)
+    @food3 = Food.create(name: 'fufu', measurement_unit: 'Litter', price: '1 hour', user_id: @user.id)
+    @food2 = Food.create(name: 'achu', measurement_unit: 'cm', price: '1 hour', user_id: @user.id)
+
+    @recipe = Recipe.create(name: 'Ekwang', description: 'Cameroonian Food', preparation_time: '1 hour',
+                            cooking_time: '1 hour', user_id: @user.id)
+    @recipe = Recipe.create(name: 'Koki', description: 'Cameroonian Food', preparation_time: 10, cooking_time: 10,
+                            public: false, user_id: @user.id)
+    @recipe = Recipe.create(name: 'Chigala', description: 'Kenian Food', preparation_time: '1 hour',
+                            cooking_time: '1 hour', user_id: @user1.id)
+  end
+
+  context 'Visit public recipe' do
+    it 'should show public recipe' do
+      visit public_recipes_path
+      expect(page).to have_content('Ekwang')
+    end
+    it 'should not show private recipe' do
+      visit public_recipes_path
+      expect(page).not_to have_content('Koki')
+    end
+    it 'requires login' do
+      visit recipes_path
+      expect(page).to have_content('You need to sign in or sign up before continuing.')
+    end
+  end
+  context 'Visit private recipe' do
+    it 'should show private recipe' do
```

Comment:
Greetings, @MahmoudBakr23 We sincerely appreciate you taking the time to examine our work; on my end, all the tests pass. I have no idea what the problem might be.

Reply:
Hi @MahmoudBakr23,

We really appreciate your kind and honest review. It warms my heart that you appreciate the work we put into the project because we really did.
We are working on optimizing that test suite as you requested.

Cheers.